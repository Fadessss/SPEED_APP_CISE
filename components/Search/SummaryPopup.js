import React from 'react'
import styles from './search.module.css';

// Function component for the summary popup which displays the selected result's details.
const SummaryPopup = ({ selectedResult, setShowPopup, showPopup }) => {

  // The component returns a popup box with details if 'showPopup' is true. Otherwise, it returns null.
  return (
    // Ternary operator that checks if 'showPopup' is true.
    showPopup ? (
        // If 'showPopup' is true, a div with class name 'styles.popup' is returned.
        <div className={styles.popup}>
          {/* The close button in the top right corner of the popup. Clicking this button will set 'showPopup' to 'false', hiding the popup */}
          <button className={styles.closeButton} onClick={() => setShowPopup(false)}>X</button>
          {/* The title of the selected result is displayed along with its year of publication and authors. */}
          <h2>{selectedResult.title} - {selectedResult.year} - {selectedResult.authors}</h2>
          {/* A summary paragraph which prints out details about the selectedResult. */}
          <p>This is the summary, about 200 words long. It provides a brief summary of the data related to the claim within the journal article.
            <br /><br />
            {/* An example sentence displaying usage of the selected result's properties. */}
            Eg. {selectedResult.title} demonstrates evidence which {selectedResult.result}s that {selectedResult.SEpractice} {selectedResult.claim}.
            <br /><br />
            {/* Additional details about the selectedResult. */}
            The article was written by {selectedResult.authors} in {selectedResult.year}, and was published in {selectedResult.journal}.
          </p>
        </div> ) : null  // If 'showPopup' is false, null is returned, resulting in no output
  );
}

// The function component 'SummaryPopup' is exported for use in other parts of the application.
export default SummaryPopup;