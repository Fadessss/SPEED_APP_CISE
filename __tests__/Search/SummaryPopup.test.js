// Import testing utilities for rendering and interaction
import { render, fireEvent } from '@testing-library/react';

// Import the component to be tested
import SummaryPopup from '../../components/Search/SummaryPopup';

// Define mock data for populating the SummaryPopup props
const selectedResult = {
    title: "TestTitle",
    yearOfPublication: "2022",
    authors: "TestAuthor",
    resultOfEvidence: "TestEvidence",
    SEPractice: "TestSEPractice",
    claim: "TestClaim",
    journalOrConferenceName: "TestConName"
};

// Begin describing the test suite for the SummaryPopup component
describe("SummaryPopup", () => {
    // Test case: It should render the popup when showPopup is true
    it("should render the popup when showPopup is true", () => {
        // Create a mock function for setShowPopup
        const setShowPopup = jest.fn();
        
        // Render the SummaryPopup component with provided data and function
        const { asFragment } = render(
            <SummaryPopup selectedResult={selectedResult} setShowPopup={setShowPopup} showPopup={true}/>
        );
        
        // Take a snapshot of the rendered component for comparison
        expect(asFragment()).toMatchSnapshot();
    });

    // Test case: It should not render the popup when showPopup is false
    it("should not render the popup when showPopup is false", () => {
        // Create a mock function for setShowPopup
        const setShowPopup = jest.fn();
        
        // Render the SummaryPopup component with provided data and function
        const { asFragment } = render(
            <SummaryPopup selectedResult={selectedResult} setShowPopup={setShowPopup} showPopup={false}/>
        );
        
        // Take a snapshot of the rendered component for comparison
        expect(asFragment()).toMatchSnapshot();
    });

    // Test case: It should call setShowPopup function when the close button is clicked
    it("should call setShowPopup function when close button is clicked", () => {
        // Create a mock function for setShowPopup
        const setShowPopup = jest.fn();
        
        // Render the SummaryPopup component with provided data and function
        const { getByText } = render(
            <SummaryPopup selectedResult={selectedResult} setShowPopup={setShowPopup} showPopup={true}/>
        );

        // Simulate a click on the 'X' close button
        fireEvent.click(getByText('X'));

        // Expect that setShowPopup function has been called
        expect(setShowPopup).toHaveBeenCalled();
    });
});
