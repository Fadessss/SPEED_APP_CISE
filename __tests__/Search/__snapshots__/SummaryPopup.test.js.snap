// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SummaryPopup should not render the popup when showPopup is false 1`] = `<DocumentFragment />`;

exports[`SummaryPopup should render the popup when showPopup is true 1`] = `
<DocumentFragment>
  <div
    class="popup"
  >
    <button
      class="closeButton"
    >
      X
    </button>
    <h2>
      TestTitle - 2022 - TestAuthor
    </h2>
    <p>
      This is the summary. It provides a brief summary of the data related to the claim within the journal article.
      <br />
      <br />
      Eg. TestTitle demonstrates evidence which TestEvidences that TestSEPractice TestClaim.
      <br />
      <br />
      The article was written by TestAuthor in 2022, and was published in TestConName.
    </p>
  </div>
</DocumentFragment>
`;
